crates/shaders/src/compile/mod.rs

```diff
@@ -18,8 +18,12 @@ use crate::types::{BindType, BindingInfo, WorkgroupBufferInfo};
 
 #[derive(Error, Debug)]
 pub enum Error {
-    #[error("failed to parse shader: {0}")]
-    Parse(#[from] wgsl::ParseError),
+    #[error("failed to parse shader ({name}): {msg}")]
```

Comment:
It would be really nice if we could point to file and line. I understand why we can't (because we'd effectively need to handle source maps), but it would be good for DX.

OTOH, as it's only "internal" DX, that's not that important

Reply:
Fortunately `emit_to_string` prints out line numbers (so `msg` contains them) and `{name}` here is will match the name of the pipeline being compiled, e.g.:

```
Failed to create renderer: Failed to compile shader: failed to parse shader (pathtag_reduce2): error: expected identifier, found '32'
   ┌─ wgsl:87:14
   │
87 │     path_ix: 32,
   │              ^^ expected identifier
```

That said the line numbers don't actually line up with the real files because following the preprocess step and won't contain any info about the included files. So yeah, we'd need some type of source mapping (which is beyond the scope of this PR).